src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs

```diff
@@ -127,13 +128,13 @@ internal string ResolveHubName(string attributeHubName)
             // First, try the Attribute's string.
             if (!string.IsNullOrEmpty(attributeHubName))
             {
-                return attributeHubName;
+                return _nameResolver.ResolveWholeString(attributeHubName);
             }
 
             // Second, try the config's HubName
             if (!string.IsNullOrEmpty(HubName))
             {
-                return HubName;
+                return _nameResolver.ResolveWholeString(HubName);
```

Comment:
What happens if _nameResolver.ResolveWholeString fails to find an app setting to resolve? Does it return string.Empty?

We probably want to check that a result was found - if not, return the original HubName.